### PR TITLE
(IAC-668) Replace templating engine

### PR DESCRIPTION
**Pull request by michaeltlombardi**
Prior to this commit the builder relied on the EPS templating
engine which has not been actively maintained for several years
now. This made the builder fragile and added cognitive load.

This commit updates the builder to do away with the templating
engine completely, opting instead to implement a file copy for
the static files and the use of here-strings in public functions
to retrieve the text for a Puppet Resource API compliant type
and provider for each DSC resource.

TODO:

- [x] Add tests
- [x] Add comment-based help
- [x] Rebase for clearer progression
- [x] Update dependency docs in https://github.com/puppetlabs/PuppetDscBuilder/blob/9bce334d973f9ad01075c773d7263a1d5976c1db/build.ps1#L23

### CODE REVIEW

**Comment by DavidS** [comment]
Rebased it on top of this morning's work.

File: src/functions/Get-TypeContent.ps1 [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    for the Puppet Resource API.
+  #>
+  [cmdletbinding()]
+  param (
+    [OutputType([String], [String[]])]
+    [Parameter(ValueFromPipeline)]
+    [Microsoft.PowerShell.DesiredStateConfiguration.DscResourceInfo[]]$DscResource
+  )
+
+  Begin {}
```
Comment (RandomNoun7) [inline src/functions/Get-TypeContent.ps1:27]: Do we need this if there is no empty end{}?

Comment (michaeltlombardi) [reply src/functions/Get-TypeContent.ps1:27]: 🤔 I actually don't know if you can skip them altogether... I guess you can? Whoops.

File: src/internal/functions/Get-TypeParameterContent.ps1 [hunk 2]
```diff
@@ -0,0 +1,54 @@
+Function Get-TypeParameterContent {
+  <#
+  .SYNOPSIS
+    Return the text for a Puppet Resource API Type given a DSC Resouce.
```
Comment (RandomNoun7) [inline src/internal/functions/Get-TypeParameterContent.ps1:4]: This synopsis and description are exactly the same as for the Get-TypeContent function.

File: src/internal/functions/Get-TypeParameterContent.ps1 [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+    This will retrieve a DSC resource from the PSModulePath, retrieve the information
+    needed to represent the DSC resource's properties as Puppet Resource API type
+    attributes and then return the representation of that DSC resource appropriate
+    for the Puppet Resource API.
+  #>
+  [cmdletbinding()]
+  Param (
+    [OutputType([String], [String[]])]
+    [pscustomobject]$ParameterInfo
```
Comment (RandomNoun7) [inline src/internal/functions/Get-TypeParameterContent.ps1:23]: This function says it's input comes from Get-DscResourceInformation as does Get-TypeContent, but the parameter type specified here is different from that function. Is that necessary? Just not sure what type of object that cmdlet is actually putting out at this point.

Comment (michaeltlombardi) [reply src/internal/functions/Get-TypeParameterContent.ps1:23]: The `ParameterInfo` property is a NoteProperty added by `Get-DscResourceTypeInformation` whose value is an array of PSCustomObjects, but the only way to get that info is via the function or to manually construct an example object (as we do in the tests).

Comment (RandomNoun7) [reply src/internal/functions/Get-TypeParameterContent.ps1:23]: Ahhh, I forgot about that.

**Pull request merged**